nlmod/dims/time.py

```diff
@@ -130,9 +133,192 @@ def set_ds_time(
+        single value, the model will have only one stress period. The default is None.
+    nstp : int or array-like, optional
+        number of steps per stress period, stored in ds.attrs, default is 1
+    tsmult : float, optional
+        timestep multiplier within stress periods, stored in ds.attrs, default is 1.0
+
+    Returns
+    -------
+    ds : xarray.Dataset
+        model dataset with added time coordinate
+
+    """
+    logger.info(
+        "Function set_ds_time() has changed since nlmod version 0.7."
+        " For the old behavior, use `nlmod.time.set_ds_time_deprecated()`."
+    )
+
+    if time is None and perlen is None:
+        raise (ValueError("Please specify either time or perlen in set_ds_time"))
+    elif perlen is not None:
+        if time is not None:
+            msg = f"Cannot use both time and perlen. Ignoring perlen: {perlen}"
+            logger.warning(msg)
+        else:
+            if isinstance(perlen, (int, np.integer, float)):
+                perlen = [perlen]
+            time = np.cumsum(perlen)
+
+    if isinstance(time, str) or not hasattr(time, "__iter__"):
+        time = [time]
+
+    # parse start
+    if isinstance(start, (int, np.integer, float)):
+        if isinstance(time[0], (int, np.integer, float)):
+            raise (ValueError("Make sure start or time contains a valid TimeStamp"))
+        start = time[0] - pd.to_timedelta(start, "D")
+    elif isinstance(start, str):
+        start = pd.Timestamp(start)
+    elif isinstance(start, (pd.Timestamp, np.datetime64)):
+        pass
+    else:
+        raise TypeError("Cannot parse start datetime.")
+
+    # convert time to Timestamps
+    if isinstance(time[0], (int, np.integer, float)):
+        time = pd.Timestamp(start) + pd.to_timedelta(time, time_units)
+    elif isinstance(time[0], str):
+        time = pd.to_datetime(time)
+    elif isinstance(time[0], (pd.Timestamp, np.datetime64, xr.core.variable.Variable)):
+        pass
+    else:
+        raise TypeError("Cannot process 'time' argument. Datatype not understood.")
+
+    if time[0] <= start:
+        msg = (
+            "The timestamp of the first stress period cannot be before or "
+            "equal to the model start time! Please modify `time` or `start`!"
+        )
+        logger.error(msg)
+        raise ValueError(msg)
+
+    ds = ds.assign_coords(coords={"time": time})
+
+    # add steady, nstp and tsmult to dataset
+    if isinstance(steady, bool):
+        steady = int(steady) * np.ones(len(time), dtype=int)
+        if steady_start:
+            steady[0] = 1
+    ds["steady"] = ("time",), steady
+
+    if isinstance(nstp, (int, np.integer)):
+        nstp = nstp * np.ones(len(time), dtype=int)
+    ds["nstp"] = ("time",), nstp
+
+    if isinstance(tsmult, float):
+        tsmult = tsmult * np.ones(len(time))
+    ds["tsmult"] = ("time",), tsmult
+
+    if time_units == "D":
+        time_units = "DAYS"
+    ds.time.attrs["time_units"] = time_units
```

Comment:
Instead import the attributes from the dims/attributes_encodings.py?
```
from dims.attributes_encodings import dims_attrs
ds.time.attrs = dims_attrs["time"]
```

And add 
```
"time":  dict(
        name="Time",
        description="Time at the end of the stress period"),
```

Reply:
See commit below